src/main/scala/org/broadinstitute/dsde/agora/server/business/AgoraBusiness.scala

```diff
@@ -320,56 +319,58 @@ class AgoraBusiness(permissionsDataSource: PermissionsDataSource) extends LazyLo
       excludedFields = Seq.empty[String])
 
     val methodsAndConfigsFuture = findWithIds(AgoraEntity(), Some(projection), Seq(AgoraEntityType.Configuration, AgoraEntityType.Workflow), username)
+    // find public-ness
+    val aliasesOwnersPermissionsFuture = permissionsDataSource.inTransaction(_.aePerms.listAliasesOwnersPermissions)
```

Comment:
Just so that I am understanding correctly, the purpose of running `permissionsDataSource.inTransaction(_.aePerms.listAliasesOwnersPermissions)` here rather than inside the `methodsAndConfigsFuture` flatMap is to avoid running the same transaction multiple times?

Reply:
Refactored and added comments.

The purpose of running the `Future`s early is that Scala's Future immediately queue-then-runs as soon as possible. Assuming there are multiple threads available, both of the Futures should start running. Later, the `flatMap` / `map` waits for both of the Futures to complete, then runs the enclosed block.

I refactored the code so that the two Futures are clearly labeled as starting as early as possible. The block that is meant to be run once the Futures have completed has been moved into its own function.